airflow/jobs.py

```diff
@@ -1002,6 +1002,12 @@ def _execute_task_instances(self,
                     not_(DR.run_id.like(BackfillJob.ID_PREFIX + '%'))))
             .all()
         )
+        # to save on queries
```

Comment:
simpler to just do paused_dags = query....
then check if task_instance.dag_id in paused_dags for the if statement later on